tests/backuptestTriggers.go

```diff
@@ -0,0 +1,628 @@
+		appContexts := ScheduleApplications(taskName)
+		for _, ctx := range appContexts {
+			ctx.ReadinessTimeout = AppReadinessTimeout
+			namespace := GetAppNamespace(ctx, taskName)
+			bkpNamespaces = append(bkpNamespaces, namespace)
+			scheduledAppContexts = append(scheduledAppContexts, ctx)
+		}
+	}
+
+	eventData.BackupNamespaces = bkpNamespaces
+	eventData.SchedulerContext = scheduledAppContexts
+
+	return nil, "", *eventData
+}
+
+// Event to validate app but later will be changed to create handler and start data for app
+func eventValidateScheduleApplication(inputsForEventBuilder *PxBackupLongevity) (error, string, EventData) {
+	defer GinkgoRecover()
+	eventData := &EventData{}
+	_, _ = ValidateApplicationsStartData(inputsForEventBuilder.ApplicationData.SchedulerContext, ctx)
+	return nil, "", *eventData
+}
+
+// Event for backup location and cred add
+func eventAddCredentialandBackupLocation(inputsForEventBuilder *PxBackupLongevity) (error, string, EventData) {
+	defer GinkgoRecover()
+	eventData := &EventData{}
+	backupLocationMap := make(map[string]string)
+	var cloudCredUidList []string
+	var providers = GetBackupProviders()
+	var backupLocationUID string
+
+	log.InfoD("Creating cloud credentials and backup location")
+	for _, provider := range providers {
+		cloudCredUID := uuid.New()
+		cloudCredUidList = append(cloudCredUidList, cloudCredUID)
+		backupLocationUID = uuid.New()
+		credName := fmt.Sprintf("autogenerated-cred-%v", time.Now().Unix())
+		err := CreateCloudCredential(provider, credName, cloudCredUID, BackupOrgID, ctx)
+		if err != nil {
+			return err, "", *eventData
+		}
+		log.InfoD("Created Cloud Credentials with name - %s", credName)
+		customBackupLocationName := fmt.Sprintf("autogenerated-backup-location-%v", time.Now().Unix())
+		err = CreateBackupLocation(provider, customBackupLocationName, backupLocationUID, credName, cloudCredUID, getGlobalBucketName(provider), BackupOrgID, "", true)
+		if err != nil {
+			return err, "", *eventData
+		}
+		backupLocationMap[backupLocationUID] = customBackupLocationName
+		log.InfoD("Created Backup Location with name - %s", customBackupLocationName)
+
+		eventData.BackupLocationName = customBackupLocationName
+		eventData.BackupLocationUID = backupLocationUID
+	}
+
+	return nil, "", *eventData
+}
+
+// Event for Source and Dest Cluster add
+func eventAddSourceAndDestinationCluster(inputsForEventBuilder *PxBackupLongevity) (error, string, EventData) {
+	defer GinkgoRecover()
+	eventData := &EventData{}
+
+	log.Infof("Adding Clusters for backup")
+	err := CreateApplicationClusters(BackupOrgID, "", "", ctx)
+	if err != nil {
+		return err, "", *eventData
+	}
+	clusterStatus, err := Inst().Backup.GetClusterStatus(BackupOrgID, SourceClusterName, ctx)
+	if err != nil {
+		return err, "", *eventData
+	}
+	if clusterStatus != api.ClusterInfo_StatusInfo_Online {
+		return fmt.Errorf("Cluster %s is not online", SourceClusterName), fmt.Sprintf("Cluster added but not online"), *eventData
```

Comment:
log the cluster status in the error message

Reply:
Done